supfile.go

```diff
@@ -13,21 +13,66 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
+// EnvVar represents an environment variable
+type EnvVar struct {
+	Key   string
+	Value string
+}
+
+func (e EnvVar) String() string {
+	return e.Key + `=` + e.Value
+}
+
+// AsExport returns the environment variable as a bash export statement
+func (e EnvVar) AsExport() string {
+	return `export ` + e.Key + `="` + e.Value + `";`
+}
+
+// EnvList is a list of environment variables that maps to a YAML map,
+// but maintains order, enabling late variables to reference early variables.
+type EnvList []EnvVar
+
+func (e *EnvList) UnmarshalYAML(ufn func(interface{}) error) error {
+	d := []yaml.MapItem{}
+
+	err := ufn(&d)
+	if err != nil {
+		return err
+	}
+
+	*e = make(EnvList, 0, len(d))
+
+	for _, v := range d {
+		e.Set(fmt.Sprintf("%v", v.Key), fmt.Sprintf("%v", v.Value))
+	}
+
+	return nil
+}
+
+// Set key to be equal value in this list.
+func (e *EnvList) Set(key, value string) {
+	// if key exists result will be redefinition
+	*e = append(*e, EnvVar{
```

Comment:
You're appending here, thus relying on rewriting the var by `export a=b; export a=c;` order, right? Can we search and override the old value instead properly?